### PR TITLE
Edit endpoint checks integration

### PR DESCRIPTION
**Pull request by Aiky30**
### Documentation checklist

* [ ] I have updated CHANGELOG.txt if appropriate
* [ ] I have updated the release notes document if appropriate, with:
    * [ ] general notes
    * [ ] bug-fixes
    * [ ] improvements/new features
    * [ ] backwards-incompatible changes
    * [ ] required upgrade steps
    * [ ] names of contributors
* [ ] I have updated other documentation
* [ ] I have added my name to the AUTHORS file
* [ ] This PR's documentation has been approved by Daniele Procida

### CODE REVIEW

File: cms/utils/placeholder.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 # -*- coding: utf-8 -*-
+from itertools import chain
```
Comment (chaosk) [inline cms/utils/placeholder.py]: Move after `from collections...`

File: cms/utils/placeholder.py [hunk 2]
```diff
@@ -363,3 +365,25 @@ def get_declared_placeholders_for_obj(obj):
     if not hasattr(obj, 'get_template'):
         raise NotImplementedError('%s should implement get_template' % obj.__class__.__name__)
     return get_placeholders(obj.get_template())
+
+
+def run_placeholder_container_checks(container, user):
+    """
+    Get all of the placeholders from the parents and
```
Comment (chaosk) [inline cms/utils/placeholder.py]: all of placeholders linked to container? I'd also rename container to source here, as that's the name used in a bunch of places already

File: cms/utils/placeholder.py [hunk 3]
```diff
@@ -363,3 +365,24 @@ def get_declared_placeholders_for_obj(obj):
...
         raise NotImplementedError('%s should implement get_template' % obj.__class__.__name__)
     return get_placeholders(obj.get_template())
+
+
+def run_source_container_checks(container, user):
+    """
+    Get all of the placeholders attached to the source
+    object and run their registered checks
+    """
+    # If the object has a get_placehodler helper method use it,
```
Comment (chaosk) [inline cms/utils/placeholder.py]: Typo

File: cms/utils/placeholder.py [hunk 4]
```diff
@@ -363,3 +365,24 @@ def get_declared_placeholders_for_obj(obj):
     if not hasattr(obj, 'get_template'):
         raise NotImplementedError('%s should implement get_template' % obj.__class__.__name__)
     return get_placeholders(obj.get_template())
+
+
+def run_source_container_checks(container, user):
```
Comment (chaosk) [inline cms/utils/placeholder.py]: run_source_checks? And rename container to source

**Comment by FinalAngel** [comment]
Closing this for now

**Comment by Aiky30** [comment]
@FinalAngel You haven't checked with us if this should be closed, We are currently moving sites into Production with CMS4, this feature will be needed for this work piece. Can you please re open this.

**Comment by Aiky30** [comment]
Half finished solution fixed by Choask.

**Pull request closed without merging**